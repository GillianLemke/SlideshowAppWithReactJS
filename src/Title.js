import React from 'react';
//import { createStore } from 'redux';
import ReactDOM from 'react-dom';
import './index.css';

//let store = createStore()
export const add = (a,b) => a+b;
export default class Title extends React.Component {
  constructor(props) {
    super(props);
    // do we need content here?
    this.state = { title:'', type:'title', content:[] };
  }
  PageInfo() {
    return $.getJSON('google.com/slide')
      .then((data) => {
        // how can we get data for individual variables?
        this.setState({ title:data.results })
      });
  }
  render() {
    return <div><button className="Next-Button" onClick={this.props.nextClick}>NEXT</button></div>;
  }

}
