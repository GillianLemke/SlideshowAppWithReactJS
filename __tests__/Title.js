import React from 'react'
import Title, {add} from '../src/Title.js'
import {shallow} from 'enzyme' // could also mount
import toJson from 'enzyme-to-json'

describe('Title', () => {
  it('renders', () => {
    const wrapper = shallow(<Title title='' type='' contents={[]} nextClick={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('adds two numbers', () => {
    const sum = add(2,2)
    expect(sum).toEqual(4)
  });
  it('responds to next click', () =>{
    const nextListener = jest.fn()
    const wrapper = shallow(<Title title='' type='' contents={[]} nextClick={nextListener} />)
    wrapper.find('.Next-Button').simulate('click') // also could do event ex. target, value
    expect(nextListener).toHaveBeenCalled()
  });

});
