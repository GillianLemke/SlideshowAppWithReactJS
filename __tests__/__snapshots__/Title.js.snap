exports[`Title renders 1`] = `
<div>
  <button
    className="Next-Button"
    onClick={[Function]}>
    NEXT
  </button>
</div>
`;
